contracts/Comptroller.sol

```diff
@@ -1021,6 +1036,44 @@ contract Comptroller is ComptrollerV3Storage, ComptrollerInterface, ComptrollerE
         allMarkets.push(CToken(cToken));
     }
 
+
+    /**
+      * @notice Set the given borrow limits for the given cToken markets
+      * @dev Admin or borrowLimitGuardian function to set the borrow limits
+      * @param cTokens The addresses of the markets (tokens) to change the borrow limits for
+      * @param newBorrowLimits The new borrow limit values in underlying (maximum borrowing) to be set
+      */
+    function _setMarketBorrowLimits(CToken[] calldata cTokens, uint256[] calldata newBorrowLimits) external {
+    	require(msg.sender == admin || msg.sender == borrowLimitGuardian, "only admin or borrow limit guardian"); 
+
+        uint numMarkets = cTokens.length;
+        uint numLimits = newBorrowLimits.length;
+
+        require(numMarkets != 0 && numMarkets == numLimits, "invalid input");
+
+        for(uint8 i = 0; i < numMarkets; i++) {
+            borrowLimits[address(cTokens[i])] = newBorrowLimits[i];
```

Comment:
Strangely (or for the better?) this allows setting limits for tokens that aren't listed / don't exist?

Reply:
Allows for setting a market borrow limit before listing it. I see this as a plus. It could be in the same governance call. I don't really see a significant downside here.